### PR TITLE
[fix]: [CDS-80771]: Added permissions for InputSets CRUD

### PR DESCRIPTION
**Pull request by vtxorxwitty**
## Describe your changes
- Added new permissions for InputSets CRUD.
- Assigned permissions to appropriate users as applicable.
- Did changes to InputSet APIs according to new InputSet permissions.
- All the pipeline service changes consuming InputSets permissions are behind FF.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)

### CODE REVIEW

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by sahilhindwani-harness** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1724 at r4](https://reviewable.io/reviews/harness/harness-core/54008#-NhBgbfyBus9iA2_X0Ol:-NhBgbfyBus9iA2_X0Om:b1dgw5q) ([raw file](https://github.com/harness/harness-core/blob/8c0a62c3fba12716972a5ec9277885971ce94de5/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1724)):*
> ```YAML
>     name: Create or Edit InputSet
>     status: ACTIVE
>     includeInAllRoles: false
> ```

I think it should not be set to false? Can you please check

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger uts

**Review by vtxorxwitty — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: 15 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1724 at r4](https://reviewable.io/reviews/harness/harness-core/54008#-NhBgbfyBus9iA2_X0Ol:-NiYeghQ6w3M94Er9Wwf:b-rn69z9) ([raw file](https://github.com/harness/harness-core/blob/8c0a62c3fba12716972a5ec9277885971ce94de5/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1724)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

I think it should not be set to false? Can you please check
</blockquote></details>

Checked on this. We have this field set to false for every permission defined.

**Comment by vtxorxwitty** [comment]
trigger uts

**Review by reetika04 — APPROVED** [review]

**Comment by reetika04** [comment]
Looks good to me.

**Comment by vtxorxwitty** [comment]
trigger ut1

**Comment by vtxorxwitty** [comment]
trigger ut8

**Comment by vtxorxwitty** [comment]
trigger ut9

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 1 of 4 files at r3, 1 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: 15 file(s) to review.

**Comment by vtxorxwitty** [comment]
trigger ut1

**Comment by vtxorxwitty** [comment]
trigger review

**Comment by vtxorxwitty** [comment]
trigger ut1

**Comment by vtxorxwitty** [comment]
trigger checkstylecodeformat

**Comment by vtxorxwitty** [comment]
trigger pmd

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger ut4

**Comment by vtxorxwitty** [comment]
trigger ut5

**Comment by vtxorxwitty** [comment]
trigger ut6

**Comment by vtxorxwitty** [comment]
trigger ut5

**Comment by vtxorxwitty** [comment]
trigger ut5

**Comment by sahilhindwani-harness** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1729 at r7](https://reviewable.io/reviews/harness/harness-core/54008#-NidLf-s3TW_bqwyxMD0:-NidLf-t16Z-gGIp5ok3:blgg210) ([raw file](https://github.com/harness/harness-core/blob/a54a92d1226e0dcfb555db5b8cee73d9707d0f7f/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1729)):*
> ```YAML
>   - identifier: core_inputset_edit
>     name: Create or Edit InputSet
>     status: ACTIVE
> ```

status should be inactive

File: access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml [hunk 1]
```diff
@@ -1723,4 +1723,28 @@ permissions:
     status: ACTIVE
     includedInAllRoles: false
     allowedScopeLevels:
-      - account
\ No newline at end of file
+      - account
+  - identifier: core_inputset_edit
+    name: Create or Edit InputSet
+    status: ACTIVE
```
Comment (sahilhindwani-harness) [inline access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:1729]: INACTIVE

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]

**Review by vtxorxwitty — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: 15 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1729 at r7](https://reviewable.io/reviews/harness/harness-core/54008#-NidLf-s3TW_bqwyxMD0:-NidO0mp3cf6hcrxd33l:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a54a92d1226e0dcfb555db5b8cee73d9707d0f7f/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1729)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

status should be inactive
</blockquote></details>

Done.
___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1729 at r7](https://reviewable.io/reviews/harness/harness-core/54008#gh-1384715402:-NidO16_7212-TXQj3Lh:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a54a92d1226e0dcfb555db5b8cee73d9707d0f7f/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1729)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

INACTIVE
</blockquote></details>

Done.

**Review by Jimit09Gandhi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1729 at r7](https://reviewable.io/reviews/harness/harness-core/54008#gh-1384715402:-Njf5AJP8Awvnz1Tnqqx:b-qst4er) ([raw file](https://github.com/harness/harness-core/blob/a54a92d1226e0dcfb555db5b8cee73d9707d0f7f/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1729)):*
<details><summary><i>Previously, vtxorxwitty (Ved Tambat) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Dont see this changed to InActive yet.
___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1729 at r7](https://reviewable.io/reviews/harness/harness-core/54008#-NidLf-s3TW_bqwyxMD0:-Njf5ErvDCYxYsKZzikI:b-qst4er) ([raw file](https://github.com/harness/harness-core/blob/a54a92d1226e0dcfb555db5b8cee73d9707d0f7f/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1729)):*
<details><summary><i>Previously, vtxorxwitty (Ved Tambat) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Dont see this changed to InActive yet.

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: Some of the checks are too old. 15 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/inputset/InputSetResourceImpl.java` line 122 at r9](https://reviewable.io/reviews/harness/harness-core/54008#-Njf9eXb204mhzWRHsNu:-Njf9eXb204mhzWRHsNv:b5kdyrk) ([raw file](https://github.com/harness/harness-core/blob/ee6a53ccd47bf0a826395944fa2acb8a52035ebe/platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/inputset/InputSetResourceImpl.java#L122)):*
> ```Java
>     return ImmutableMap.of(ScopeLevel.ACCOUNT, EnumSet.of(BY_RESOURCE_TYPE_INCLUDING_CHILD_SCOPES),
>         ScopeLevel.ORGANIZATION, EnumSet.of(BY_RESOURCE_TYPE_INCLUDING_CHILD_SCOPES), ScopeLevel.PROJECT,
>         EnumSet.of(BY_RESOURCE_IDENTIFIER, BY_RESOURCE_TYPE));
> ```

Do we want to support granular BY\_RESOURCE\_IDENTIFIER on input set?

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: Some of the checks are too old. 15 file(s) to review. 2 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/resources/InputSetResourcePMSImpl.java` line 160 at r9](https://reviewable.io/reviews/harness/harness-core/54008#-NjfCD0096RJcKhru-DP:-NjfCD032vQVRR5ntE2G:b5sgwi4) ([raw file](https://github.com/harness/harness-core/blob/ee6a53ccd47bf0a826395944fa2acb8a52035ebe/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/resources/InputSetResourcePMSImpl.java#L160)):*
> ```Java
>     } else {
>       accessControlClient.checkForAccessOrThrow(ResourceScope.of(accountId, orgIdentifier, projectIdentifier),
>           Resource.of("PIPELINE", pipelineIdentifier), PipelineRbacPermissions.PIPELINE_VIEW);
> ```

Since the permission check will now be based on Input set instead of pipeline, what about permissions for existing users?
When is migration planned for it?

**Review by vtxorxwitty — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: Some of the checks are too old. 15 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1729 at r7](https://reviewable.io/reviews/harness/harness-core/54008#-NidLf-s3TW_bqwyxMD0:-NjfVZ41-C3Gjyynj-WL:bccd2kv) ([raw file](https://github.com/harness/harness-core/blob/a54a92d1226e0dcfb555db5b8cee73d9707d0f7f/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1729)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Dont see this changed to InActive yet.
</blockquote></details>

We've decided to not have this permissions inactive @Jimit09Gandhi 
___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1729 at r7](https://reviewable.io/reviews/harness/harness-core/54008#gh-1384715402:-NjfViEmDN9jgmXVaiG4:b-5i3fhe) ([raw file](https://github.com/harness/harness-core/blob/a54a92d1226e0dcfb555db5b8cee73d9707d0f7f/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1729)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Dont see this changed to InActive yet.
</blockquote></details>

We've decided to have the permissions to be active. @Jimit09Gandhi 
___
*[`pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/resources/InputSetResourcePMSImpl.java` line 160 at r9](https://reviewable.io/reviews/harness/harness-core/54008#-NjfCD0096RJcKhru-DP:-NjfVwvI3Ouk2iBLmyo-:bqwbg2y) ([raw file](https://github.com/harness/harness-core/blob/ee6a53ccd47bf0a826395944fa2acb8a52035ebe/pipeline-service/service/src/main/java/io/harness/pms/ngpipeline/inputset/resources/InputSetResourcePMSImpl.java#L160)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Since the permission check will now be based on Input set instead of pipeline, what about permissions for existing users?
When is migration planned for it?
</blockquote></details>

We will be doing a migration for permissions. It is planned to be picked up by @rohitk-h under the jira https://harness.atlassian.net/browse/CDS-83302 

Until then we would not enable the FF and PIPELINE permissions will be used to as of now. 
___
*[`platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/inputset/InputSetResourceImpl.java` line 122 at r9](https://reviewable.io/reviews/harness/harness-core/54008#-Njf9eXb204mhzWRHsNu:-NjfWXqzCwj_6dRbzFJE:bb77uf2) ([raw file](https://github.com/harness/harness-core/blob/ee6a53ccd47bf0a826395944fa2acb8a52035ebe/platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/resourceclient/inputset/InputSetResourceImpl.java#L122)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Do we want to support granular BY\_RESOURCE\_IDENTIFIER on input set?
</blockquote></details>

Yes

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 4 of 12 files at r2, 2 of 9 files at r5, 1 of 1 files at r6, 3 of 4 files at r8, 4 of 4 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: Some of the checks are too old. 15 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 4 of 12 files at r2, 1 of 4 files at r3, 2 of 9 files at r5, 1 of 1 files at r6, 3 of 4 files at r8, 4 of 4 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: Some of the checks are too old.

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger uts

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: 1 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 1756 at r10](https://reviewable.io/reviews/harness/harness-core/54008#-NjlIC9d3PLDFNpqKso0:-NjlIC9d3PLDFNpqKso1:bee1cr6) ([raw file](https://github.com/harness/harness-core/blob/13b14b2ec71eb991ec3838dbb5248e2b14e48acc/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L1756)):*
> ```YAML
>       - project
>   - identifier: core_inputset_view
>     name: View InputSet
> ```

Add core\_inputset\_view to access-control/service/src/main/resources/io/harness/accesscontrol/roles/privileged-roles.yml

**Comment by vtxorxwitty** [comment]
trigger uts

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 1 of 1 files at r10, 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54008)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by vtxorxwitty** [comment]
trigger ut6

**Comment by vtxorxwitty** [comment]
trigger ut9

**Comment by vtxorxwitty** [comment]
trigger uts

**Comment by vtxorxwitty** [comment]
trigger uts

**Pull request merged**